### PR TITLE
Add tutorial for defining generic sensor resource

### PR DESCRIPTION
**Pull request by SeppoTakalo**
Signed-off-by: Seppo Takalo <seppo.takalo@nordicsemi.no>

### CODE REVIEW

File: include/net/nrf91_lwm2m_app.rst [hunk 1]
```diff
@@ -44,6 +44,132 @@ By default, the library uses LTE-M for connecting and doesn't utilize a bootstra
...
+This section will walk through a process of creating a custom LwM2M resource. For most application, the
+business logic may be implemented inside custom resources, which are either proprietary to this application
+or may be following some external specification, like IPSO objects. Either way, they extend the resource tree
+by defining own objects on top of what LwM2M already defines.
+
+Before defining a proprietary resource, it is a good practice to check whether similar functionality is
+already defined in `LwM2M Object and Resource Registry`_.
+If there is already such definition you might want to use that, instead of defining your own.
+
+Please note that Zephyr's LwM2M library has only a limited support for some IPSO objects, so extending beyound those,
```
Comment (VeijoPesonen) [inline include/net/nrf91_lwm2m_app.rst:59]: And supports only LwM2M v1.0.2 at the moment.

File: include/net/nrf91_lwm2m_app.rst [hunk 2]
```diff
@@ -44,6 +44,132 @@ By default, the library uses LTE-M for connecting and doesn't utilize a bootstra
...
+.. code:: c
+
+   int init_resource(void)
+   {
+      lwm2m_engine_create_obj_inst("3300/0");
+      lwm2m_engine_register_read_callback("3300/0/5700", read_cb);
+      return 0;
+   }
+
+API is quite self explanatory. First we need to create an object. This causes internal engine to allocate memory and
```
Comment (VeijoPesonen) [inline include/net/nrf91_lwm2m_app.rst:166]: "API is quite self explanatory" might sound a bit demeaning. Maybe something like "To put in other words" or something similar.

**Review by VeijoPesonen — APPROVED** [review]
Maybe fix those two and then merge.

File: include/net/nrf91_lwm2m_app.rst (hunk 1 above)
Comment (SeppoTakalo) [reply include/net/nrf91_lwm2m_app.rst:59]: For IPSO objects, or custom resources, this is not relevant.

File: include/net/nrf91_lwm2m_app.rst (hunk 2 above)
Comment (SeppoTakalo) [reply include/net/nrf91_lwm2m_app.rst]: I changed to
> As show above, first we need to create an object and then register callbacks for specific resources.

**Pull request merged**